src/components/ErrorSummary.tsx

```diff
@@ -1,18 +1,86 @@
 import React, { FunctionComponent, ReactNode } from "react";
+import { IFormError } from "../lib/formValidator";
+
+interface FormErrorSummaryProps {
+  errors: IFormError[];
+}
+
+interface FormErrorSummaryLinkProps {
+  href: string;
+  errorMessage: string;
+}
+
+interface FieldErrorListProps {
+  href: string;
+  errorMessages: string[];
+}
+
+interface FieldErrorMessageProps {
+  errorMessage: string;
+}
 
 interface ErrorSummaryProps {
   children: ReactNode;
 }
 
-export const ErrorSummary: FunctionComponent<ErrorSummaryProps> = ({
+export const FormErrorSummary: FunctionComponent<FormErrorSummaryProps> = ({
+  errors,
+}: FormErrorSummaryProps) => (
+  <>
+    {errors && errors.length > 0 && (
+      <ErrorSummary>
+        {errors.map((field) =>
+          field.errorMessages.map((errorMessage, index) => (
+            <FormErrorSummaryLink
+              key={`${field.fieldId}-${index}`}
+              href={`#${field.fieldId}`}
+              errorMessage={error}
+            />
+          ))
+        )}
+      </ErrorSummary>
+    )}
+  </>
+);
+
+const FormErrorSummaryLink: FunctionComponent<FormErrorSummaryLinkProps> = ({
+  href,
+  errorMessage,
+}: FormErrorSummaryLinkProps) => (
+  <li>
+    <a href={href}>{errorMessage}</a>
+  </li>
+);
+
+export const FieldErrorList: FunctionComponent<FieldErrorListProps> = ({
+  href,
+  errorMessages,
+}: FieldErrorListProps): JSX.Element => (
+  <>
+    {errorMessages.map((message, index) => (
+      <FieldErrorMessage errorMessage={message} key={`${href}-${index}`} />
+    ))}
+  </>
+);
+
+const FieldErrorMessage: FunctionComponent<FieldErrorMessageProps> = ({
+  errorMessage,
+}: FieldErrorMessageProps) => (
+  <span className="govuk-error-message">
+    <span className="govuk-visually-hidden">Error:</span>
+    {errorMessage}
+  </span>
+);
+
+const ErrorSummary: FunctionComponent<ErrorSummaryProps> = ({
   children,
 }: ErrorSummaryProps): JSX.Element => (
   <div
     className="govuk-error-summary"
     aria-labelledby="error-summary-title"
     role="alert"
     tabIndex={-1}
-    data-module="govuk-error-summary"
+    data-module="govuk-errorMessage-summary"
```

Comment:
Oh no, is this a casualty of find and replace 😂

Reply:
Oh crumbs 🤦🏻 This is me not yet fully knowing how to use WebStorm's rename refactor.  Thanks for catching!